ibc-testkit/src/context.rs

```diff
@@ -40,34 +37,22 @@ where
     H: TestHost,
     HostClientState<H>: ClientStateValidation<MockIbcStore<S>>,
 {
+    /// The main store of the context.
+    pub main_store: S,
```

Comment:
Why this `main_store` field has been added? Couldn't benefit from the `store` field under the `MockIbcStore` instead?

Reply:
This is needed because IBC needs an ICS23 proof for `commitment_prefix` as well.

https://github.com/cosmos/ibc-rs/blob/b0ea4eab7247c50519099f86f9d2c42db51eed07/ibc-clients/ics07-tendermint/src/client_state/common.rs#L204-L205

So TmClient requires two Merkle proofs (chained together), one for the IBC storage root at `commitment_prefix` and the other one for the value at the IBC storage path.

Ref: [ADR-09](https://github.com/cosmos/ibc-rs/blob/rano/testkit/adr-009/docs/architecture/adr-009-revamp-testkit.md#ics23-compatible-proof-generation)

Reply:
Thank you for providing the context. That's fine then.